src/ie9-ajax.js

```diff
@@ -51,15 +53,24 @@ function ie9Ajax(method, url, opts) {
       }
     };
 
-    function handleError() {
-      reject(new Error('There was an error with the XDR'));
+    function handleError(error) {
+      logger.error(`There was an error with the XDR\n${JSON.stringify(error, null, '  ')}`);
     }
 
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        logger.error('Please ensure Shopify JavaScript Buy SDK is run over https');
+      } else {
+        logger.error(`There was an error when running XDR\n${error.message}`);
```

Comment:
I think that's what @tessalt said would crash IE and not really show much that's useful?